pkg/collector/systemperf_collector.go

```diff
@@ -22,23 +42,73 @@ func (collector *SystemPerfCollector) GetName() string {
 
 // Collect implements the interface method
 func (collector *SystemPerfCollector) Collect() error {
-	if err := utils.CreateKubeConfigFromServiceAccount(); err != nil {
-		return err
+	metric, err := metrics.NewForConfig(collector.kubeconfig)
+	if err != nil {
+		return fmt.Errorf("metrics for config error: %w", err)
+	}
+
+	// Node Metrics collector
+	nodeMetrics, err := metric.MetricsV1beta1().NodeMetricses().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("node metrics error: %w", err)
+	}
+
+	noderesult := make([]NodeMetrics, 0)
+
+	for _, nodeMetric := range nodeMetrics.Items {
+		cpuQuantity := nodeMetric.Usage.Cpu().MilliValue()
+		memQuantity, ok := nodeMetric.Usage.Memory().AsInt64()
+		if !ok {
+			return err
+		}
+
+		nm := NodeMetrics{
+			NodeName:    nodeMetric.Name,
+			CPUUsage:    cpuQuantity,
+			MemoryUsage: memQuantity,
+		}
+
+		noderesult = append(noderesult, nm)
+	}
+	jsonNodeResult, err := json.Marshal(noderesult)
+	if err != nil {
+		return fmt.Errorf("marshall node metrics to json: %w", err)
 	}
 
-	output, err := utils.RunCommandOnContainer("kubectl", "top", "nodes")
+	collector.data["nodes"] = string(jsonNodeResult)
+
+	// Pod Metrics collector
+	podMetrics, err := metric.MetricsV1beta1().PodMetricses(metav1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
```

Comment:
what's the `es` in `metric.MetricsV1beta1().PodMetricses`? Is there no metric.MetricsV1beta1().PodMetrics() function?

Reply:
Indeed, yep, no funct name what you without `es`, its https://pkg.go.dev/k8s.io/metrics/pkg/client/clientset/versioned/typed/metrics/v1beta1